core/modules/mod.rs

```diff
@@ -353,6 +359,7 @@ impl ModuleType {
 pub struct ModuleSource {
   pub code: ModuleSourceCode,
   pub module_type: ModuleType,
+  pub code_cache: Option<Cow<'static, [u8]>>,
```

Comment:
Is using `'static` here required by V8 API? Seems very restrictive 😬

Reply:
No, it's not required by V8 API. Will make it less restrictive.

Reply:
After thinking about this some more, I'm not sure this would be useful. If the embedder loads the cache from some external source (e.g. file on disk or a database) - they will just load the cache and hand over the owned `Vec<u8>` when creating `ModuleSource`. There is no reason for the implementor to keep the cache around after creating `ModuleSource`. It would just complicate the cache lifetime management since `ModuleSource` has to outlive the cache. It would make sense for `ModuleSource` to just own the cache with `Vec<u8>`.

Providing code cache as `&[u8]` would be useful if the implementor had the cache as part of their binary (in which case the lifetime is `'static`).

Do you have some specific use case in mind?

Reply:
So my thinking is that if we were to provide it for some user modules (reading form DENO_DIR) we'd have to leak the cache data to transmute it to `'static` lifetime. Using owned vector seems much more approachable and would prevent potential leaks if you spawn workers and provide code cache for them but then the worker is terminated but the memory has already been leaked.

Reply:
With `Cow<'static, [u8]>` we would be able to do both. In the case of user modules that I'm targeting - it will be owned vector read from DENO_DIR. I think the `'static` reference can be used in the future for providing code cache for builtin modules (or maybe for user modules from `deno compile` binary). We could just change it to `Vec<u8>` for now since that's what we need for user modules.

Or are you specifically thinking about the use case where multiple workers are loading the same modules, so we'd want to share the same code cache across all workers?

Reply:
I'm taking about the latter, but for the first pass it's fine if we just copy the code cache data. 

So it's fine really for now.